src/types/app_context.go

```diff
@@ -11,12 +9,12 @@ type AppContext struct {
 }
 
 // GetServiceContext returns a ServiceContext for the service found at the given directory base
-func (a *AppContext) GetServiceContext(serviceLocation string) (*ServiceContext, error) {
-	serviceConfig, err := NewServiceConfig(serviceLocation)
+func (a *AppContext) GetServiceContext(serviceRole string, serviceData ServiceData) (*ServiceContext, error) {
+	serviceConfig, err := NewServiceConfig(path.Join(a.Location, serviceData.Location))
 	return &ServiceContext{
-		Dir:        path.Base(serviceLocation),
-		Location:   serviceLocation,
+		Role:       serviceRole,
 		Config:     serviceConfig,
 		AppContext: a,
+		AppData:    &serviceData,
```

Comment:
Would it make more sense to call this ServiceData? After all that's what it represents.